### PR TITLE
mgr/dashboard: added delete and activation actions

### PR DESCRIPTION
**Pull request by ivoalmeida**
Fixes: https://tracker.ceph.com/issues/64355

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

[Screencast from 2024-02-08 12-44-56.webm](https://github.com/ceph/ceph/assets/1071297/f18573bc-24f5-4546-80df-9f806dc53463)

### CODE REVIEW

**Comment by afreen23** [comment]
Hi, I tested this locally and deletion and activation-deactivation works fine, verified from cli as well. 🎉 

One minor thing, may be we can bold the directory we are deleting (kind of highlighting the deletion object)
Screenshot from 2024-02-08 19-18-25

For example as done in other  views like of ceph fs deletion we are bolding the deleted object so may be we can do it here.
Screenshot from 2024-02-08 19-21-36
And for rgw users views:
image

**Review by afreen23 — APPROVED** [review]
LGTM, the functionality working e2e as I tried locally, hence approving, thanks!
The minor comment above can be taken care now or later as its a different component not touched in this PR..

**Comment by ivoalmeida** [comment]
> Hi, I tested this locally and deletion and activation-deactivation works fine, verified from cli as well. 🎉
> 
> One minor thing, may be we can bold the directory we are deleting (kind of highlighting the deletion object) Screenshot from 2024-02-08 19-18-25 But its coming bolded when I create snapshot with some full directory name image I assume this is the way css or component is configured..

hi @afreen23 

thanks for the feedback. Actually this should not have been the selected path. I fixed that. The message should just ask about the selected snapshot schedule. The reason for this is that you can have more than one snap schedule per path.

This is how it looks now:
image

**Comment by ivoalmeida** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/cephfs-snapshot-schedule.service.ts [hunk 1]
```diff
@@ -19,14 +19,38 @@ export class CephfsSnapshotScheduleService {
     return this.http.post(`${this.baseURL}/snapshot/schedule`, data, { observe: 'response' });
   }
 
-  update(data: Record<string, any>): Observable<any> {
+  update({fs, path, ...rest}: Record<string, any>): Observable<any> {
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/api/cephfs-snapshot-schedule.service.ts:22]: what is `rest` refers to here?

**Review by nizamial09 — APPROVED** [review]
Instant approval because of mario and luigi.

LGTM! Thank you @ivoalmeida

**Comment by nizamial09** [comment]
```
[lint:html    ] 
[lint:html    ] [htmllint] found 0 errors out of 212 files
[lint:html    ] All files pass linting
[lint:prettier] src/app/shared/api/cephfs-snapshot-schedule.service.ts
[lint:eslint  ] 
[lint:eslint  ] Linting "ceph-dashboard"...
[lint:eslint  ] 
[lint:eslint  ] All files pass linting.
[lint:eslint  ] 

Try running 'npm run fix' to fix some linting errors. Some errors might need a manual fix.

> ceph-dashboard@0.0.0 i18n:extract
> ng extract-i18n --output-path src/locale --progress=false
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/cephfs-snapshot-schedule.service.ts (hunk 1 above)
Comment (ivoalmeida) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/api/cephfs-snapshot-schedule.service.ts]: I am destructuring the arg object to grab the properties I need and the rest (which is only the remaining properties) I pass in the body of the http request.

More info below:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters
https://dev.to/coderjay06/how-to-destructure-nested-javascript-objects-1dcl

**Comment by ivoalmeida** [comment]
jenkins test windows

**Comment by ivoalmeida** [comment]
jenkins test windows

**Comment by ivoalmeida** [comment]
jenkins test windows

**Comment by nizamial09** [comment]
jenkins test windows

**Comment by ivoalmeida** [comment]
jenkins test windows

**Comment by ivoalmeida** [comment]
jenkins test windows

**Comment by cloudbehl** [comment]
jenkins test make check

**Pull request merged**